cmd/operator-sdk/new/cmd.go

```diff
@@ -368,8 +379,13 @@ func verifyFlags() error {
 		return fmt.Errorf("value of --helm-chart-version can only be used with --type=helm and --helm-chart")
 	}
 
-	if operatorType == projutil.OperatorTypeGo && (len(apiVersion) != 0 || len(kind) != 0) {
-		return fmt.Errorf("operators of type Go do not use --api-version or --kind")
+	if operatorType == projutil.OperatorTypeGo {
+		if len(apiVersion) != 0 || len(kind) != 0 {
+			return fmt.Errorf("operators of type Go do not use --api-version or --kind")
+		}
+		if !makeVendor && projutil.DepManagerType(depManager) == projutil.DepManagerDep {
+			return fmt.Errorf("--vendor cannot be used with --dep-manager=dep")
```

Comment:
With this check `operator-sdk new --vendor=true --dep-manager=dep` will result in the condition not being true and the error `--vendor cannot be used with --dep-manager=dep` will not be printed. 
Meaning `--vendor` can be used with `--dep-manager=dep`.
Unless I'm completely missing something?

Reply:
The error message should say `"--vendor cannot be set to false with --dep-manager=dep". On further consideration, we might want to skip checking `--vendor` entirely if `--dep-manager=dep` because dep requires `vendor`. WDYT?

Reply:
> On further consideration, we might want to skip checking --vendor entirely if --dep-manager=dep

In that case `operator-sdk new --vendor=false --dep-manager=dep` would silently ignore `--vendor` and we'd create a vendor directory anyway. I think I prefer the message that says you can't disable vendor with dep.

Reply:
I would not silently ignore it, but print out an error msg and still create the vendor, error msg something like:
` "dep requires vendor directory; ignoring vendor=false flag" `